indexdata.go

```diff
@@ -93,6 +93,13 @@ type indexData struct {
 
 	// repository indexes for all the files
 	repos []uint16
+
+	// public[i] is true iff repoMetaData[i] is a public repository.
+	public []bool
```

Comment:
We could save some space by using a packed bitmap here instead of 1 byte per repo.

Reply:
We expect public to very small, less than 1000 for sure, so they size of this slice should be less than 1kb per compound shard.

Reply:
EDIT: I guess we could use an uint16 mask instead of []bool. It would make the code less readable but it wouldn't be too bad.